scipy/special/gen_harmonic.h

```diff
@@ -0,0 +1,231 @@
+#include <cmath>
+#include <cstdint>
+#include <xsf/zeta.h>  // for xsf::cephes::zeta(a, n)
+#include "sf_error.h"
+
+
+//
+// Compute sum_{i=m}^{n} i**-a.
+//
+// This function assumes 1 <= m <= n and `a` is finite.
+//
+static inline double
+sum_powers(int64_t m, int64_t n, double a)
+{
+    double sum = 0.0;
+    if (a >= 0) {
+        for (int64_t i = n; i >= m; --i) {
+            sum += std::pow(i, -a);
+        }
+    }
+    else {
+        for (int64_t i = m; i <= n; ++i) {
+            sum += std::pow(i, -a);
+        }
+    }
+    return sum;
+}
+
+//
+// Compute
+//
+//   sum_{i=1}^{n} i**-a
+//
+// This is the generalized harmonic number H_{n, a} [1].
+//
+// * If n < 1, NAN is returned with SF_ERROR_DOMAIN set.
+// * If a > 1 and not too close to 1, the sum is computed using
+//   the Hurwitz zeta function.  Otherwise it is computed as a simple
+//   sum of the powers.
+//
+// [1] https://en.wikipedia.org/wiki/Harmonic_number#Generalized_harmonic_numbers
+//
+static inline double
+gen_harmonic(int64_t n, double a)
+{
+    if (n < 1) {
+        sf_error("_gen_harmonic", SF_ERROR_DOMAIN,
+                 "n >= 1 is required, but got n = %" PRId64, n);
+        return NAN;
+    }
+    if (std::isnan(a)) {
+        if (n == 1) {
+            // IEEE: pow(1.0, nan) is 1.0.
+            return 1.0;
+        }
+        return NAN;
+    }
+    if (std::isinf(a)) {
+        if (a > 0) {
+            // a = +inf
+            return 1.0;
+        }
+        else {
+            // a = -inf
+            if (n == 1) {
+                // IEEE: pow(1.0, inf) = 1
+                return 1.0;
+            }
+            else {
+                return INFINITY;
+            }
+        }
+    }
```

Comment:
Is there a reason not to do this? Same can be done in the normalized version.
```suggestion
    if (n == 1) {
        // IEEE: pow(1.0, _) is 1.0.
        return 1.0;
    }
    if (std::isnan(a)) {
        return NAN;
    }
    if (std::isinf(a)) {
        if (a > 0) {
            // a = +inf
            return 1.0;
        }
        else {
            // a = -inf
            return INFINITY;
        }
    }
```